src/backend/distributed/shared_library_init.c

```diff
@@ -670,6 +670,16 @@ RegisterCitusConfigVariables(void)
 		0,
 		NULL, NULL, NULL);
 
+	DefineCustomStringVariable(
+		"citus.cluster_name",
+		gettext_noop("Which cluster this coordinator sends queries to"),
+		NULL,
+		&CurrentCluster,
+		"default",
+		PGC_USERSET,
```

Comment:
I'm probably overthinking this, but how does `BACKEND` sound?

Going through the options in `guc.h`:
- `POSTMASTER` options can only be changed by restarting the cluster
- `SIGHUP` options can be changed by sending `HUP` to the postmaster/backend
- `BACKEND` can be changed but remains constant from the point of view of any given backend. It must be changed by editing the config file or specifying it in `PGOPTIONS` at startup.

It seems that,
+ `POSTMASTER` perfectly fits the use-case we currently envision, you setup a coordinator in a different cluster and let it run, but it might be too inflexible? It's harder to test this if you have to reconfigure and restart the cluster to change `cluster_name`.
+ The backends accept the `SIGHUP` before processing every packet so it's possible for the value of `citus.cluster_name` to change between planning and execution (Between the `Bind` and `Execute` messages) That sounds too difficult to have to think about while coding.
+ However, `BACKEND` maintains a constant cluster during the lifetime of any backend, and lets you change the cluster future backends use. This might be a useful operation for administrators? I can see it being useful for testing Citus.

If you're worried about users being able to set it, we can use `SU_BACKEND` which disables the startup packet option unless you're a superuser.